discovery-provider/tests/test_track_upload_challenge.py

```diff
@@ -0,0 +1,155 @@
+from datetime import datetime
+import redis
+from src.models import User, Block, Track
+from src.utils.db_session import get_db
+from src.challenges.track_upload_challenge import track_upload_challenge_manager
+from src.challenges.challenge_event_bus import ChallengeEventBus, ChallengeEvent
+from src.utils.config import shared_config
+
+REDIS_URL = shared_config["redis"]["url"]
+
+
+def test_track_upload_challenge(app):
+
+    redis_conn = redis.Redis.from_url(url=REDIS_URL)
+
+    # create user
+    with app.app_context():
+        db = get_db()
+
+    block1 = Block(blockhash="0x1", number=1)
+    block2 = Block(blockhash="0x2", number=20000000)
+    block3 = Block(blockhash="0x3", number=20000001)
+    user = User(
+        blockhash="0x1",
+        blocknumber=1,
+        txhash="xyz",
+        user_id=1,
+        handle="TestHandle",
+        handle_lc="testhandle",
+        is_current=True,
+        created_at=datetime.now(),
```

Comment:
nit, probably should pick somewhat more reflective created_at / updated_at dates. you can do `- timedelta(days=X)),`